x-pack/plugins/observability/server/lib/rules/custom_threshold/register_custom_threshold_rule_type.ts

```diff
@@ -75,6 +75,7 @@ export function thresholdRuleType(
     threshold: schema.arrayOf(schema.number()),
     comparator: oneOfLiterals(Object.values(Comparator)),
     timeUnit: schema.string(),
+    id: schema.maybe(schema.string()),
```

Comment:
With this change, we are adding an id to the rule's saved object and our create/update API just for using it in UI, which I don't think is a good idea. Can we handle it only on the UI side without adding a field to the rule's saved object?

<img src="https://github.com/elastic/kibana/assets/12370520/37112a33-4177-468a-8a83-76f35da662bd" width=300 />

Reply:
id is only used in UI and isn't a required field when let's say user is using an API.